src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationService/HealthCertificateValidationService.swift

```diff
@@ -172,4 +212,557 @@ final class HealthCertificateValidationService: HealthCertificateValidationProvi
+		}
+		
+		Log.info("Successfully validated invalidation rules: \(private: invalidationRulesValidations). Proceed with combined rule validation...", log: .vaccination)
+
+		let combinedRuleValidations = acceptanceRulesValidations + invalidationRulesValidations
+		proceedWithRuleInterpretation(
+			combinedRuleValidations: combinedRuleValidations,
+			completion: completion
+		)
+		
+	}
+	
+	private func proceedWithRuleInterpretation(
+		combinedRuleValidations: [ValidationResult],
+		completion: @escaping (Result<HealthCertificateValidationReport, HealthCertificateValidationError>) -> Void
+	) {
+		if combinedRuleValidations.allSatisfy({ $0.result == .passed }) {
+			// all rules has to be .passed
+			Log.info("Successfully combined rules: \(private: combinedRuleValidations). Validation result is: validationPassed. Validation complete.", log: .vaccination)
+			completion(.success(.validationPassed))
+		} else if combinedRuleValidations.allSatisfy({ $0.result == .open }) {
+			// all rules has to be .open
+			Log.info("Successfully combined rules: \(private: combinedRuleValidations). Validation result is: validationOpen. Validation complete.", log: .vaccination)
+			completion(.success(.validationOpen(combinedRuleValidations)))
+		} else {
+			// At least one rule should contain now .fail
+			Log.info("Successfully combined rules: \(private: combinedRuleValidations). Validation result is: validationFailed. Validation complete.", log: .vaccination)
+			completion(.success(.validationFailed(combinedRuleValidations)))
+		}
+	}
+	
+	// MARK: - Acceptance Rules
+		
+	private func downloadAcceptanceRule(
+		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
+	) {
+		client.getDCCRules(
+			eTag: store.acceptanceRulesCache?.lastValidationRulesETag,
+			isFake: false,
+			ruleType: .acceptance,
+			completion: { [weak self] result in
+				guard let self = self else {
+					Log.error("Could not create strong self", log: .vaccination)
+					completion(.failure(.ACCEPTANCE_RULE_CLIENT_ERROR))
+					return
+				}
+				
+				switch result {
+				case let .success(packageDownloadResponse):
+					self.acceptanceRuleDownloadingSuccessHandler(
+						packageDownloadResponse: packageDownloadResponse,
+						completion: completion
+					)
+	
+				case let .failure(error):
+					self.acceptanceRuleDownloadingFailureHandler(
+						error: error,
+						completion: completion
+					)
+				}
+			}
+		)
+	}
+	
+	private func acceptanceRuleDownloadingSuccessHandler(
+		packageDownloadResponse: PackageDownloadResponse,
+		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
+	) {
+		Log.info("Successfully received acceptance rules package. Proceed with eTag verification...", log: .vaccination)
+
+		guard let eTag = packageDownloadResponse.etag else {
+			Log.error("ETag of package is missing. Return with failure.", log: .vaccination)
+			completion(.failure(.ACCEPTANCE_RULE_JSON_ARCHIVE_SIGNATURE_INVALID))
```

Comment:
Is this the correct error to return in this case?

Reply:
I remembered for another story that this case was mapped on some kind of this error. But anyway, I added a new error case specially for this case